### PR TITLE
Fix support for translations

### PR DESCRIPTION
**Pull request by Diontsoumas**
API requests for creating/updating a translation were failing due to a missing path and a wrong http method used. Added tests and updated the README file.

### CODE REVIEW

File: txlib/api/tests/test_translation.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+@pytest.fixture(scope='module', autouse=True)
+def auto_clean_registry():
+    """Run the test and the remove the `http_handler` entry from
+    the registry."""
+    yield
+    clean_registry()
+
+
+class TestTranslationModel():
+    """Test the functionality of the Resource model."""
```
Comment (dbendilas) [inline txlib/api/tests/test_translation.py]: ~Resource~ Translation

File: txlib/api/translations.py [hunk 2]
```diff
@@ -7,7 +8,21 @@ class Translation(BaseModel):
...
 
     _path_to_item = 'project/%(project_slug)s/resource/%(slug)s/translation/%(lang)s'  # noqa
-    _path_to_collection = None
+    _path_to_collection = 'project/%(project_slug)s/resource/'\
+                          '%(slug)s/translation/%(lang)s'
 
     writable_fields = {'content'}
     url_fields = {'project_slug', 'slug', 'lang'}
+
+    def _create(self, **kwargs):
```
Comment (dbendilas) [inline txlib/api/translations.py]: I just saw that this method is not necessary. The base `_create()` method, defined in `base.py` has exactly the same body.

Comment (Diontsoumas) [reply txlib/api/translations.py]: @dbendilas is not the same, the one in `base.py` makes a `POST` request, this one makes a `PUT` request. The translations endpoint needs the latter.

Comment (dbendilas) [reply txlib/api/translations.py]: Ah, ok. I didn't see that, I don't think it's that obvious.

We could do one of the following 2 things:

1. Amend the docstring of the new `_create()` method to explicitly state that it's a `PUT` operation.

2. Support both options on the base class and override when necessary, e.g.

```
def _create(self, **kwargs):
    self._create_with_post(**kwargs)

def _create_with_post(self, **kwargs):
    path, kwargs = self._get_create_params(**kwargs)
    return self._http.post(path, json.dumps(kwargs))

def _create_with_put(self, **kwargs):
    path, kwargs = self._get_create_params(**kwargs)
    return self._http.put(path, json.dumps(kwargs))

def _get_create_params(self, **kwargs):
    kwargs = dict(kwargs)
    path = self._construct_path_to_collection()

    # Use the fields for which we have values
    for field in self.writable_fields:
        try:
            value = getattr(self, field)
            kwargs[field] = value
        except AttributeError:
            pass

    return path, kwargs
```

```
# translations.py
def _create(self, **kwargs):
    return self._create_with_put(**kwargs)
```

**Pull request merged**